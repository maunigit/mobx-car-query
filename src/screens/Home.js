import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import { observer, inject } from 'mobx-react';
import { Button, List } from 'react-native-paper';
import globalStyles from '../styles/Style';

@inject('store')
@observer
export default class Home extends React.Component {
  //Screen navigation
  static navigationOptions = {
    title: 'Home',
  };

  disableModel = () => {
    if(this.props.store.make!=''&&this.props.store.year>0){
      return false;
    }
    return true;
  }

  disableTrim = () => {
    if(this.props.store.model!=''){
      return false;
    }
    return true;
  }

  reset = () =>{
    this.props.store.make='';
    this.props.store.year=0;
    this.props.store.model='';
    this.props.store.trim_model_id='';
  }

  //Show filters
  render() {
    return (
      <View style={globalStyles.homeContainer}>
        <TouchableOpacity >
          <List.Item button onPress={() => this.props.navigation.navigate('Year')} 
            title={'Year'} description={'Select a year'}
            left={props => <List.Icon {...props} icon="calendar-range"/>}/>
        </TouchableOpacity>
        <TouchableOpacity >
          <List.Item button onPress={() => this.props.navigation.navigate('Make')}
            title={'Make'} description={'Select a make'}
            left={props => <List.Icon {...props} icon="car"/>}/>
        </TouchableOpacity>           
        <TouchableOpacity >
          <List.Item button onPress={() => this.props.navigation.navigate('Model')} 
            title={'Model'} description={'Avaiable models of a make'} disabled={this.disableModel()} 
            left={props => <List.Icon {...props} icon="car-side"/>}/>
        </TouchableOpacity>   
        <TouchableOpacity >
          <List.Item button onPress={() => this.props.navigation.navigate('Trim')} 
            title={'Trim'} description={'Avaiable trims of a model'} disabled={this.disableTrim()} 
            left={props => <List.Icon {...props} icon="engine-outline"/>}/>
        </TouchableOpacity>
        <View style={globalStyles.button}>
          <Button
            color="orange"
            mode="contained"
            onPress={() => this.props.navigation.navigate('Detail')}>
            show details
          </Button>
          <Button
            color="red"
            mode="contained"
            onPress={() => this.reset()}>
            reset
          </Button>
        </View>        
      </View>    
    );
  }
}
