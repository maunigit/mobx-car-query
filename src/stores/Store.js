import { observable } from 'mobx';

//Place to store information
class Store {
  @observable data = [];
  @observable make = '';
}

export default new Store();
