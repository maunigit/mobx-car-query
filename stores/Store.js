import { observable, action } from 'mobx';

class Store {
  @observable data = [];
  @observable model = '';
  @observable country = '';
}

export default new Store();
