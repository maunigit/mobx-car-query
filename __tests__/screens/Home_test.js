import React from 'react';
import Home from '../../src/screens/Home';
import Store from "../../src/stores/Store";
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const store = Store; 
configure({adapter: new Adapter()});

it('Home renders correctly', () => {     
  const wrapper = shallow(<Home.wrappedComponent store={store}/>);
  expect(wrapper).toMatchSnapshot();
});

it('Home renders listitem-make', () => {     
  const wrapper = shallow(<Home.wrappedComponent store={store}/>); 
  expect(wrapper.exists('#li-make')).toEqual(true);
});

it('Home renders listitem-year', () => {     
  const wrapper = shallow(<Home.wrappedComponent store={store}/>); 
  expect(wrapper.exists('#li-year')).toEqual(true);
});

it('Home renders listitem-model disabled', () => {     
  const wrapper = shallow(<Home.wrappedComponent store={store}/>);
  const listItemElement = wrapper.find('#li-model');  
  expect(listItemElement.props().disabled).toBe(true);
});

it('Home renders listitem-trim disabled', () => {     
  const wrapper = shallow(<Home.wrappedComponent store={store}/>);
  const listItemElement = wrapper.find('#li-trim');  
  expect(listItemElement.props().disabled).toBe(true);
});

it('Home renders button-details disabled', () => {     
  const wrapper = shallow(<Home.wrappedComponent store={store}/>);
  const listItemElement = wrapper.find('#button-details');  
  expect(listItemElement.props().disabled).toBe(true);
});

it('Home renders button-reset', () => {     
  const wrapper = shallow(<Home.wrappedComponent store={store}/>);
  expect(wrapper.exists('#button-reset')).toEqual(true);
});
