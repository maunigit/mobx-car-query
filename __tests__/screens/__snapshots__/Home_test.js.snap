// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "textAlign": "center",
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <withTheme(List.Item)
      button={true}
      description="Select a make"
      descriptionStyle={
        Object {
          "color": "black",
          "fontSize": 16,
        }
      }
      id="li-make"
      left={[Function]}
      onPress={[Function]}
      title="Make"
      titleStyle={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <withTheme(List.Item)
      button={true}
      description="Select a year"
      descriptionStyle={
        Object {
          "color": "black",
          "fontSize": 16,
        }
      }
      id="li-year"
      left={[Function]}
      onPress={[Function]}
      title="Year"
      titleStyle={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <withTheme(List.Item)
      button={true}
      description="Avaiable models of a make"
      descriptionStyle={
        Object {
          "color": "gray",
          "fontSize": 16,
        }
      }
      disabled={true}
      id="li-model"
      left={[Function]}
      onPress={[Function]}
      title="Model"
      titleStyle={
        Object {
          "color": "gray",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
  >
    <withTheme(List.Item)
      button={true}
      description="Avaiable trims of a model"
      descriptionStyle={
        Object {
          "color": "gray",
          "fontSize": 16,
        }
      }
      disabled={true}
      id="li-trim"
      left={[Function]}
      onPress={[Function]}
      title="Trim"
      titleStyle={
        Object {
          "color": "gray",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    />
  </TouchableOpacity>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-around",
      }
    }
  >
    <withTheme(Button)
      color="orange"
      disabled={true}
      icon="check"
      id="button-details"
      mode="contained"
      onPress={[Function]}
    >
      show details
    </withTheme(Button)>
    <withTheme(Button)
      color="lightgray"
      icon="close"
      id="button-reset"
      mode="contained"
      onPress={[Function]}
    >
      reset
    </withTheme(Button)>
  </View>
</View>
`;
